test/backend/certificates.spec.js

```diff
@@ -228,20 +224,42 @@ describe('Certificates', function() {
         saveSetting('USERS', [{username, accessToken}]);
         saveSetting('CERTIFICATE_LAST_UPDATED', new Date());
 
-        const {cert, key} = fakeCerts;
         // Set max tries to 2.
         setCertificatesSettings('MAX_TRIES_COUNT', 2);
         assert.equal(ServerCA.count, 0, 'Make sure initially count is zero.');
 
         // Check that certs were created by timeoutFetchAndSaveCerts.
-        setRenewalJob(5000); // Renew the certificates in 5s.
+        setRenewalJob({timeout: 5000}); // Renew the certificates in 5s.
 
-        // Check that certificates were renewed.
         setTimeout(() => {
             assert.equal(ServerCA.count, 2, 'Made two tries to renew.');
             assert.deepEqual(getCerts(), {});
             done();
         }, 7000);
     }).timeout(10000);
 
+    it('setRenewalJob - Restarts the https server.', (done) => { ServerCA.start(201, fakeCerts);
+        const {cert, key} = fakeCerts;
+        // Save bogus certs.
+        saveCertsLocally({cert: 'cert', key: 'key'});
+        setTimeout(() => {
+            assert.deepEqual(getCerts(), {cert: 'cert', key: 'key'});
+        }, 200);
+        // Don't create certs because we want to see if renewing replaces the
+        // above bogus certs.
+        ServerUnderTest = new Servers({startHTTPS: true, createCerts: false});
+        // Set a spy on ServerUnderTest.httpsServer.restart
+        saveSetting('USERS', [{username, accessToken}]);
+        ServerUnderTest.httpsServer.restart = chai.spy(() => {});
+        // Renew the certificates in 1s.
+        setRenewalJob({server: ServerUnderTest.httpsServer, timeout: 1000});
+        setTimeout(() => {
+            const {restart} = ServerUnderTest.httpsServer;
+            assert.equal(ServerCA.count, 1, 'Made a call to renew.');
+            assert.deepEqual(getCerts(), {cert, key}, 'Real certificates appeared.');
+            expect(restart).to.have.been.called();
```

Comment:
maybe add a `expect(restart).not.to.have.been.called()` before this setTimeout just to make sure that it wasn't called before the delay